server/tests-py/test_tests.py

```diff
@@ -103,61 +113,222 @@ def test_tests_dict_ordering_assumptions_and_helpers(self):
+                                        [
+                                            ("id", 2),
+                                            (
+                                                "jsonb_col",
+                                                dict([("age", 8), ("name", "Rawkz")]),
+                                            ),
+                                        ]
+                                    ),
+                                ],
+                            ),
+                            ("thing2", dict([("a", 1), ("b", 2), ("c", 3)])),
+                        ]
+                    ),
+                )
+            ]
+        )
 
         # NOTE: use str() to actually do a stong equality comparison, comparing
         # types. Only works because str() on dict seems to have a canonical
         # ordering.
         assert str(relevant_ordered_result) == str(relevant_ordered_result_expected)
 
         # Demonstrate equality on different mixes of trees of ordered and unordered dicts:
-        assert CommentedMap([("a", "a"), ("b", "b")]) ==         dict([("b", "b"), ("a", "a")])
-        assert CommentedMap([("a", "a"), ("b", "b")]) != CommentedMap([("b", "b"), ("a", "a")])
-        assert         dict([           ("x", CommentedMap([("a", "a"), ("b", CommentedMap([("b1", "b1"), ("b2", "b2")]))])), ("y","y"),]) == \
-               CommentedMap([("y","y"), ("x",         dict([("a", "a"), ("b", CommentedMap([("b1", "b1"), ("b2", "b2")]))])), ])
+        assert CommentedMap([("a", "a"), ("b", "b")]) == dict([("b", "b"), ("a", "a")])
+        assert CommentedMap([("a", "a"), ("b", "b")]) != CommentedMap(
+            [("b", "b"), ("a", "a")]
+        )
+        assert dict(
+            [
+                (
+                    "x",
+                    CommentedMap(
+                        [("a", "a"), ("b", CommentedMap([("b1", "b1"), ("b2", "b2")]))]
+                    ),
+                ),
+                ("y", "y"),
+            ]
+        ) == CommentedMap(
+            [
+                ("y", "y"),
+                (
+                    "x",
+                    dict(
+                        [("a", "a"), ("b", CommentedMap([("b1", "b1"), ("b2", "b2")]))]
+                    ),
+                ),
+            ]
+        )
 
     def test_tests_ordering_differences_correctly_ignored(self, hge_ctx):
         """
         We don't care about ordering of stuff outside the selection set e.g. JSON fields.
         """
-        check_query_f(hge_ctx, 'test_tests/user_can_query_jsonb_values_filter_okay_orders.yaml', 'http')
+        check_query_f(
+            hge_ctx,
+            "test_tests/user_can_query_jsonb_values_filter_okay_orders.yaml",
+            "http",
+        )
+
+    # Re-use setup and teardown from where we adapted this test case:
+    @classmethod
+    def dir(cls):
+        return "queries/graphql_query/permissions"
+
+
+@usefixtures("per_class_tests_db_state")
+class TestTests3:
```

Comment:
@jberryman have added the tests for `validate_http_anyq_with_allowed_responses` here. I also ran the code formatter on this file and hence it seems like there are a lot of changes. Just leaving this here so that you don't get lost.

Reply:
Also, I don't think if I understand the usage of fixtures very well. Hope that is not an issue here.

Reply:
(I can't seem to comment inline above, so just posting here)

Is it easy to revert the formatting changes, especially lines 149-152? I think the reformatted version obscures what we're doing here (that's why I took the time to format it this way in the first place). 

If this is very difficult to do I don't want to hold up the merge.